File: src/main/java/com/indiepost/service/StaticPageService.java
package com.indiepost.service;

import com.indiepost.dto.StaticPageDto;
import com.indiepost.enums.Types;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Created by jake on 17. 3. 5.
 */
public interface StaticPageService {

    Long save(StaticPageDto staticPageDto);

    void update(StaticPageDto staticPageDto);

    StaticPageDto findById(Long id);

    StaticPageDto findBySlug(String slug);

    Page<StaticPageDto> find(Pageable pageable);

    Page<StaticPageDto> find(Types.PostStatus status, Pageable pageable);

    void delete(StaticPageDto staticPageDto);

    void deleteById(Long id);

    Long count();

    void bulkUpdateStatus(List<Long> ids, Types.PostStatus status);

    void bulkDeleteByIds(List<Long> ids);

    void emptyTrash();
}

File: src/main/kotlin/com/indiepost/service/StaticPageService.kt
package com.indiepost.service

import com.indiepost.dto.StaticPageDto
import com.indiepost.enums.Types
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

/**
 * Created by jake on 17. 3. 5.
 */
interface StaticPageService {

    fun save(dto: StaticPageDto): Long

    fun update(dto: StaticPageDto)

    fun findById(id: Long): StaticPageDto?

    fun findBySlug(slug: String): StaticPageDto?

    fun find(pageable: Pageable): Page<StaticPageDto>

    fun find(status: Types.PostStatus, pageable: Pageable): Page<StaticPageDto>

    fun delete(staticPageDto: StaticPageDto)

    fun deleteById(id: Long)

    fun count(): Long

    fun bulkUpdateStatus(ids: List<Long>, status: Types.PostStatus)

    fun bulkDeleteByIds(ids: List<Long>)

    fun emptyTrash()
}
